bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -499,7 +519,7 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 		return ctrl.Result{}, err
 	}
 
-	cloudJoinData, err := cloudinit.NewNode(&cloudinit.NodeInput{
+	nodeInput := &cloudinit.NodeInput{
```

Comment:
I agree, this could be done eventually, but I'm not sure if now is the right thing to do this in the scope of this PR, which is already quite big and opened since a while. I'd rather consider Ignition support as experimental for now, where cloud-init is still a primary format used, so Ignition plugs to the functionality supported by cloud-init. I'm happy to refactor it later on into a dedicated type. Do note, that once we have a dedicated type, adding any other user data format should support all the features we put there.

Reply:
@binchenX do you mind if I mark it as resolved?

Reply:
yes, please.